semeio/workflows/localisation/local_script_lib.py

```diff
@@ -207,34 +210,327 @@ def activate_gen_param(
+                        # active regions.
+                        v = scaling_values[index]
+                        sumv += v
+                        nval += 1
+            if nval > 0:
+                scaling_values_smooth[index0] = sumv / nval
+    return scaling_values_smooth
+
+
+def set_missing_code_for_inactive_grid_cells(grid, region_parameter, missing_code):
+    data_size_active = grid.get_num_active()
+    data_size_all = grid.get_global_size()
+    mask_parameter = np.zeros(data_size_all, dtype=np.uint16)
+    for index in range(data_size_active):
+        global_index = grid.get_global_index(active_index=index)
+        mask_parameter[global_index] = 1
+    region_parameter[mask_parameter == 0] = missing_code
+
+
+def apply_segment(
+    row_scaling,
+    data_size,
+    grid,
+    region_param_dict,
+    active_segment_list,
+    scaling_factor_list,
+    smooth_range_list,
+    corr_name,
+    log_level=LogLevel.OFF,
+):
+    """
+    Purpose: Use region numbers and list of scaling factors per region to
+                   create scaling factors per active .
+                   Input dictionary with keyword which is correlation group name,
+                   where values are numpy vector with region parameters
+                   for each grid cell in ERTBOX grid.
+                   A scaling factor is specified for each specified active region.
+                   Optionally also a spatial smoothing of the scaling factors
+                   can be done by specifying smooth ranges in number of
+                   grid cells in I and J direction. If this is not specified,
+                   no smoothing is done.
+                   NOTE: Smoothing is done only between active segments,
+                   and no smoothing between active segments and inactive
+                   segments or inactive grid cells.
+    """
+
+    debug_print(f"Active segments: {active_segment_list}", LogLevel.LEVEL3, log_level)
+    if scaling_factor_list is None:
+        scaling_factor_list = []
+        scaling_factor_list.append(1.0)
+        debug_print(
+            f"Scaling factors:    {scaling_factor_list}", LogLevel.LEVEL3, log_level
+        )
+
+    max_region_number_specified = 0
+    for n in active_segment_list:
+        if n > max_region_number_specified:
+            max_region_number_specified = n
+        if n < 0:
+            raise ValueError("Segment number must be non-negative integer")
+
+    region_parameter = region_param_dict[corr_name]
+    missing_code = 999
+    if missing_code in region_parameter:
+        logging.warning(
+            "Input region parameter %s has values equal to %d\n"
+            "The value is reserved as missing code internally in "
+            "the localisation script."
+        )
+    max_region_parameter = region_parameter.max()
+    if max_region_parameter < max_region_number_specified:
+        raise ValueError(
```

Comment:
I tried to avoid the need for access to ERT instance in configuration. That is why I have not read the region parameters, only the file names there. Some checks and file reading depending on e.g grid is therefore postponed to where I have full access to the ERT instance.